reporting/src/main/java/org/wso2/testgrid/reporting/surefire/TestResult.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.reporting.surefire;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The summarized surefire test results.
+ */
+public class TestResult {
+    String totalTests = "?";
+    String totalFailures = "?";
+    String totalErrors = "?";
+    String totalSkipped = "?";
```

Comment:
Access specifier can be private. Also since this is holds a count as a numeric value and count could not be less than 0, can't we declare these counts as int and use -1 to indicate values not initialized?

Reply:
Access modifier is set as package-private because these are accessed by SurefireReporter class.

The variable type is a string because the underlying maven-surefire-reporter dependency actually returns these as strings (ref: https://github.com/apache/maven-surefire/blob/master/surefire-report-parser/src/main/java/org/apache/maven/plugins/surefire/report/SurefireReportParser.java#L152).
Since we do not need this as an int, I kept the String type as it is.

Question mark, ?, is set because we are going to display this in the email report. We can opt for a "-1" string as well here.